modules/mtl/arrow-mtl-data/src/main/kotlin/arrow/mtl/AccumT.kt

```diff
@@ -0,0 +1,211 @@
+package arrow.mtl
+
+import arrow.Kind
+import arrow.core.Either
+import arrow.core.Tuple2
+import arrow.core.identity
+import arrow.core.toT
+import arrow.higherkind
+import arrow.typeclasses.Applicative
+import arrow.typeclasses.Functor
+import arrow.typeclasses.Monad
+import arrow.typeclasses.Monoid
+
+/**
+ * Alias that represent stateful computation of the form `(S) -> Tuple2<S, A>` with a result in certain context `F`.
+ */
+typealias AccumTFun<S, F, A> = (S) -> Kind<F, Tuple2<S, A>>
+
+/**
+ * Alias that represents wrapped stateful computation in context `F`.
+ */
+typealias AccumTFunOf<S, F, A> = Kind<F, AccumTFun<S, F, A>>
```

Comment:
Oh and remember to wrap the correct places with AndThen when you invoke this function. (Unless you invoke it inside `F`, then you both cannot and need not use AndThen (because F is then responsible for stacksafety)

Reply:
thanks, cool. i removed the wrapper & tests are green now! still need to look into the `AndThen`'s

Reply:
Yup, basically everywhere where you have a combinator (take in `AccumT` return `AccumT`, or similar) and you use `AccumT { s -> runAccumT(s) }` in any form without first `flatMapping` inside `F` needs to be wrapped in `AndThen`

Reply:
i've added `AndThen´ in 3 places. i did not understand how to use it in the other places where `runAccumT` is used. think i  need some more direction here ...